### PR TITLE
DM-40404: Remove newinstall documentation

### PR DESCRIPTION
**Pull request by mwittgen**

### CODE REVIEW

File: index.rst [hunk 1]
```diff
@@ -54,7 +54,6 @@ Alternative distributions and installation methods:
 - :doc:`install/docker`
 - :doc:`Installing from source with lsstsw <install/lsstsw>`
 - `CernVM FS <https://sw.lsst.eu>`_ (contributed by CC-IN2P3)
-- :doc:`Installing with newinstall.sh <install/newinstall>`
```
Comment (roceb) [inline index.rst:57]: Should we mention `lsstinstall` here?

**Review by roceb — APPROVED** [review]
Looks good to me

File: index.rst (hunk 1 above)
Comment (mwittgen) [reply index.rst:57]: lsstinstall is mentioned in the section above as recommended. 
newinstall was already an alternative installation method.

File: install/package-development.rst [hunk 2]
```diff
@@ -2,7 +2,7 @@
 Building a package with the installed Science Pipelines stack
 #############################################################
 
-You can build packages alongside an LSST Science Pipelines stack that you have installed with :doc:`newinstall.sh <newinstall>`, :doc:`lsstsw <lsstsw>`, or :doc:`Docker <docker>`.
+You can build packages alongside an LSST Science Pipelines stack that you have installed with :doc:`lsstsw <lsstsw>` or :doc:`Docker <docker>`.
```
Comment (ktlim) [inline install/package-development.rst:5]: Why leave lsstinstall out of this list and the one below?

**Review by ktlim — APPROVED** [review]

File: install/package-development.rst (hunk 2 above)
Comment (mwittgen) [reply install/package-development.rst]: Indeed. Fixed.

**Pull request merged**